api/client/container/stop.go

```diff
@@ -40,16 +41,35 @@ func NewStopCommand(dockerCli *client.DockerCli) *cobra.Command {
 
 func runStop(dockerCli *client.DockerCli, opts *stopOptions) error {
 	ctx := context.Background()
+	timeout := time.Duration(opts.time) * time.Second
 
+	var waitAll sync.WaitGroup
+	var printLock sync.Mutex
+	var errLock sync.Mutex
 	var errs []string
+
+	var maxParallel int32 = 1000
+	//A buffered channel can be used like a semaphore, for instance to limit throughput. https://golang.org/doc/effective_go.html#channels
+	var sem = make(chan int, maxParallel)
 	for _, container := range opts.containers {
-		timeout := time.Duration(opts.time) * time.Second
-		if err := dockerCli.Client().ContainerStop(ctx, container, &timeout); err != nil {
-			errs = append(errs, err.Error())
-		} else {
-			fmt.Fprintf(dockerCli.Out(), "%s\n", container)
-		}
+		sem <- 1 // Wait for active queue sem to drain.
+		waitAll.Add(1)
+		go func(toStop string) {
+			defer waitAll.Done()
+
+			if err := dockerCli.Client().ContainerStop(ctx, toStop, &timeout); err != nil {
+				errLock.Lock()
+				errs = append(errs, err.Error())
+				errLock.Unlock()
+			} else {
+				printLock.Lock()
+				fmt.Fprintf(dockerCli.Out(), "%s\n", toStop)
```

Comment:
I was hoping users can get the response as soon as the container is stopped. Imagine that we have 10 containers, the 3rd container will fail to stop after 10 seconds with an error, if we decide to keep the order of output, the reset 7 containers have to wait until the client finish processing of the 3rd container. 
Things can go worse when we try to stop more containers. I'd like to give user a quick response instead of order.  Besides, I'm not sure if there's any scripts or command relying on the output order of `stop`.